components/ai_chat/core/browser/engine/engine_consumer_oai_unittest.cc

```diff
@@ -0,0 +1,243 @@
+  EngineConsumer::ConversationHistory history;
+
+  std::string human_input = "Which show is this catchphrase from?";
+  std::string selected_text = "This is the way.";
+  std::string assistant_input = "This is mandalorian.";
+  std::string page_content = "This is a page.";
+  bool is_video = false;
+
+  history.push_back(mojom::ConversationTurn::New(
+      mojom::CharacterType::HUMAN, mojom::ActionType::SUMMARIZE_SELECTED_TEXT,
+      mojom::ConversationTurnVisibility::VISIBLE, human_input, selected_text,
+      std::nullopt));
+
+  history.push_back(mojom::ConversationTurn::New(
+      mojom::CharacterType::ASSISTANT, mojom::ActionType::RESPONSE,
+      mojom::ConversationTurnVisibility::VISIBLE, assistant_input, std::nullopt,
+      std::nullopt));
+
+  std::string expected_human_input =
+      base::StrCat({base::ReplaceStringPlaceholders(
+                        l10n_util::GetStringUTF8(
+                            IDS_AI_CHAT_LLAMA2_SELECTED_TEXT_PROMPT_SEGMENT),
+                        {selected_text}, nullptr),
+                    "\n\n", human_input});
+
+  const std::string prompt_segment_article =
+      page_content.empty()
+          ? ""
+          : base::StrCat(
+                {base::ReplaceStringPlaceholders(
+                     l10n_util::GetStringUTF8(
+                         is_video ? IDS_AI_CHAT_LLAMA2_VIDEO_PROMPT_SEGMENT
+                                  : IDS_AI_CHAT_LLAMA2_ARTICLE_PROMPT_SEGMENT),
+                     {page_content}, nullptr),
+                 "\n\n"});
+
+  std::string date_and_time_string =
+      base::UTF16ToUTF8(TimeFormatFriendlyDateAndTime(base::Time::Now()));
+  std::string expected_system_message = base::StrCat(
+      {prompt_segment_article,
+       base::ReplaceStringPlaceholders(
+           l10n_util::GetStringUTF8(IDS_AI_CHAT_LLAMA2_SYSTEM_MESSAGE_GENERIC),
+           {date_and_time_string}, nullptr)});
+
+  auto* client = GetClient();
+  base::RunLoop run_loop;
+
+  EXPECT_CALL(*client, PerformRequest(_, _, _, _))
+      .WillOnce(
+          [&](const mojom::CustomModelOptions& model_options,
+              base::Value::List messages,
+              EngineConsumer::GenerationDataCallback,
+              EngineConsumer::GenerationCompletedCallback completed_callback) {
+            // system role is added by the engine
+            EXPECT_EQ(*messages[0].GetDict().Find("role"), "system");
+            EXPECT_EQ(*messages[0].GetDict().Find("content"),
+                      expected_system_message);
+
+            EXPECT_EQ(*messages[1].GetDict().Find("role"), "user");
+            EXPECT_EQ(*messages[1].GetDict().Find("content"),
+                      expected_human_input);
+
+            EXPECT_EQ(*messages[2].GetDict().Find("role"), "assistant");
+            EXPECT_EQ(*messages[2].GetDict().Find("content"), assistant_input);
+
+            EXPECT_EQ(*messages[3].GetDict().Find("role"), "user");
+            EXPECT_EQ(*messages[3].GetDict().Find("content"),
```

Comment:
nit: since this is just base::Value::List, you could have compared JSON output to an expected JSON string - might have been easier for you to go from OpenAI API docs to test, and less repetition / clearer testing. That's what's done in the example from ConversationAPI's unit tests